### ISSUE TITLE
blas_test_full_wilson (Failed) in feature/generic_kernel (and feature/randfft_cherry_pick)

### ISSUE THREAD

**Issue by bjoo** [lattice/quda#1070]
Just a note that while I was working on cherry picking the FFT refactor and CURAND refactor into `feature/generic_kernel` from `feature/hip-compile-fixes` I noticed that on summit the `blas_test_full_wilson` test failed...

I have mentioned this to @maddyscientist and sent her the output summary. Appears to be with some reductions: 

```[  FAILED  ] 24 tests, listed below:
2: [  FAILED  ] QUDA/BlasTest.verify/reDotProductNorm_block_half_half, where GetParam() = (4, 33)
2: [  FAILED  ] QUDA/BlasTest.verify/reDotProduct_block_half_half, where GetParam() = (4, 34)
2: [  FAILED  ] QUDA/BlasTest.verify/cDotProductNorm_block_half_half, where GetParam() = (4, 35)
2: [  FAILED  ] QUDA/BlasTest.verify/cDotProduct_block_half_half, where GetParam() = (4, 36)
2: [  FAILED  ] QUDA/BlasTest.verify/reDotProductNorm_block_half_single, where GetParam() = (5, 33)
2: [  FAILED  ] QUDA/BlasTest.verify/reDotProduct_block_half_single, where GetParam() = (5, 34)
2: [  FAILED  ] QUDA/BlasTest.verify/cDotProductNorm_block_half_single, where GetParam() = (5, 35)
2: [  FAILED  ] QUDA/BlasTest.verify/cDotProduct_block_half_single, where GetParam() = (5, 36)
2: [  FAILED  ] QUDA/BlasTest.verify/reDotProductNorm_block_half_double, where GetParam() = (6, 33)
2: [  FAILED  ] QUDA/BlasTest.verify/reDotProduct_block_half_double, where GetParam() = (6, 34)
2: [  FAILED  ] QUDA/BlasTest.verify/cDotProductNorm_block_half_double, where GetParam() = (6, 35)
2: [  FAILED  ] QUDA/BlasTest.verify/cDotProduct_block_half_double, where GetParam() = (6, 36)
2: [  FAILED  ] QUDA/BlasTest.verify/reDotProductNorm_block_single_single, where GetParam() = (7, 33)
2: [  FAILED  ] QUDA/BlasTest.verify/reDotProduct_block_single_single, where GetParam() = (7, 34)
2: [  FAILED  ] QUDA/BlasTest.verify/cDotProductNorm_block_single_single, where GetParam() = (7, 35)
2: [  FAILED  ] QUDA/BlasTest.verify/cDotProduct_block_single_single, where GetParam() = (7, 36)
2: [  FAILED  ] QUDA/BlasTest.verify/reDotProductNorm_block_single_double, where GetParam() = (8, 33)
2: [  FAILED  ] QUDA/BlasTest.verify/reDotProduct_block_single_double, where GetParam() = (8, 34)
2: [  FAILED  ] QUDA/BlasTest.verify/cDotProductNorm_block_single_double, where GetParam() = (8, 35)
2: [  FAILED  ] QUDA/BlasTest.verify/cDotProduct_block_single_double, where GetParam() = (8, 36)
2: [  FAILED  ] QUDA/BlasTest.verify/reDotProductNorm_block_double_double, where GetParam() = (9, 33)
2: [  FAILED  ] QUDA/BlasTest.verify/reDotProduct_block_double_double, where GetParam() = (9, 34)
2: [  FAILED  ] QUDA/BlasTest.verify/cDotProductNorm_block_double_double, where GetParam() = (9, 35)
2: [  FAILED  ] QUDA/BlasTest.verify/cDotProduct_block_double_double, where GetParam() = (9, 36)```

**Comment by maddyscientist** [comment]
This bug has already beed fixed in the generic/kernels branch (40bbacc7394755e4a4b5f932a6495905c5d4e1df).

**Comment by bjoo** [comment]
Verified. Thank you!
